tests/chainer_tests/links_tests/loss_tests/test_crf1d.py

```diff
@@ -0,0 +1,69 @@
+import unittest
+
+import numpy
+
+import chainer
+from chainer.backends import cuda
+from chainer import initializers
+from chainer import links
+from chainer import testing
+from chainer.testing import attr
+
+
+@testing.parameterize(*testing.product({
+    'dtype': [numpy.float16, numpy.float32, numpy.float64]
```

Comment:
Could you parameterize over initializers instead, similar to https://github.com/chainer/chainer/blob/master/tests/chainer_tests/links_tests/activation_tests/test_maxout.py#L36 ?

Reply:
`TestLinear` (for `L.Linear`) parameterizes over dtypes.
https://github.com/chainer/chainer/blob/master/tests/chainer_tests/links_tests/connection_tests/test_linear.py#L20-L21

Reply:
You're suggesting we parameterize over both? I guess that's fine too.